mkosi/__init__.py

```diff
@@ -6295,12 +6324,18 @@ def load_args(args: argparse.Namespace) -> CommandLineArguments:
     if args.skip_final_phase and args.verb != "build":
         die("--skip-final-phase can only be used when building an image using 'mkosi build'")
 
-    if args.ssh and not args.network_veth:
-        die("--ssh cannot be used without --network-veth")
+    if args.ssh and not args.base_packages:
+        die("--ssh cannot be installed without --base-packages")
+
+    if args.ssh and not args.network_veth and args.ssh_port is None:
+        die("--ssh cannot be used without --network-veth or --ssh-port")
 
     if args.ssh_timeout < 0:
         die("--ssh-timeout must be >= 0")
 
+    if args.ssh_port is not None and args.ssh_port <= 0:
```

Comment:
Seeing this `not None` check, would it maybe make sense to give `ssh_port` a default of 22, so that we always know it has a sensible value?

Reply:
ok, done